File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head is None:
            return True
        
        slow = head
        fast = slow.next

        

        while slow and fast and fast.next:
            if slow == fast:
                return True

            slow = slow.next
            fast = fast.next.next

        return False
        

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
 
public class Solution {
    public ListNode ReverseList(ListNode head) {
        return ReverseNode(head, null);
    }

    private ListNode ReverseNode(ListNode curr, ListNode prev)
    {
        if (curr == null)
        {
            return prev;
        }
        else
        {
            ListNode next = curr.next;
            curr.next = prev;

            return ReverseNode(next, curr);
        }
    }
}
